src/main/java/com/gorlah/kappabot/function/response/formatter/HtmlToPlainTextResponseFormatter.java

```diff
@@ -0,0 +1,27 @@
+package com.gorlah.kappabot.function.response.formatter;
+
+import com.gorlah.kappabot.function.response.BotResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.gorlah.kappabot.util.StandardContentTypes.HTML;
+import static com.gorlah.kappabot.util.StandardContentTypes.PLAIN_TEXT;
+
+@Component
+class HtmlToPlainTextResponseFormatter implements ResponseFormatter {
+
+    @Override
+    public Optional<String> format(BotResponse response, String contentType) {
+        return Stream.of(response)
+                .filter(resp -> HTML.equals(resp.getContentType()))
```

Comment:
Technically you could even take this a step further and just extract these predicates to a utility since you're doing this type of thing in the other `ResponseFormatter`s as well. But once again that's probably totally overkill.

Reply:
I don't think there is enough of a benefit to extract these simple predicates, either in terms of code reuse or readability.

I've been thinking about how to encapsulate all the filtering logic (top three lines of the stream) and here are some possibilities I thought of:
1. **Extract to a utility method.** I don't like this one because the method would require four arguments three of which would all be content types, thus detracting from readability.
2. **Extract an abstract class.** I don't like this either because the abstract class would require arguments be passed to its constructor, which is undesirable because it gets in the way of letting spring use the lombok constructors.
3. **Extracting an object.** I like this approach the best but I'm still on the fence about whether I like it any better than what we already have.
```java
@Value
class ResponseFilter {

    String fromContentType;
    String toContentType;

    Optional<BotResponse> filter(BotResponse response, String contentType) {
        return Stream.of(response)
                .filter(resp -> fromContentType.equals(resp.getContentType()))
                .filter(resp -> toContentType.equals(contentType))
                .findFirst();
    }
}
```
```java
private final ResponseFilter responseFilter = 
        new ResponseFilter(PLAIN_TEXT, MARKDOWN);

    @Override
    public Optional<String> format(BotResponse response, String contentType) {
        return responseFilter.filter(response, contentType)
                .map(BotResponse::getMessage);
    }
}
```

Reply:
I looked a bit closer at what you are doing here and I see now that you're not exactly doing the same thing in each of the formatters. In that case I think what you have here is fine.